File: problems/0238-product-of-array-except-self/python/solution.py
def productExceptSelf(nums):
    """Product of Array Except Self — https://leetcode.com/problems/product-of-array-except-self/"""
    n = len(nums)
    res = [1] * n
    prefix = 1
    for i in range(n):
        res[i] = prefix
        prefix *= nums[i]
    suffix = 1
    for i in range(n - 1, -1, -1):
        res[i] *= suffix
        suffix *= nums[i]
    return res

File: problems/0238-product-of-array-except-self/typescript/solution.ts
// Product of Array Except Self — https://leetcode.com/problems/product-of-array-except-self/
export function productExceptSelf(nums: number[]): number[] {
  const n = nums.length;
  const res = new Array<number>(n).fill(1);
  let prefix = 1;
  for (let i = 0; i < n; i++) {
    res[i] = prefix;
    prefix *= nums[i];
  }
  let suffix = 1;
  for (let i = n - 1; i >= 0; i--) {
    res[i] *= suffix;
    suffix *= nums[i];
  }
  return res;
}
